syntax = "proto3";

package workloads;

// Service definition for handling video streaming
service VideoStreamer {
  rpc StreamVideo (stream VideoChunk) returns (TaskResponse);
}

// Service definition for general task management
service TaskManager {
  rpc SendTask (TaskRequest) returns (TaskResponse);
  rpc RetrieveFrames (TaskRequest) returns (FrameResponse);
}

// Message for video chunk data
message VideoChunk {
  bytes data = 1;
  string processing_type = 2;
}

// Response message for general task
message TaskResponse {
  string message = 1;
  string task_id = 2;
}

// Request message for general task
message TaskRequest {
  string task_id = 1;
  string task_type = 2;
  string payload = 3;
}

// Response message for retrieved frames
message FrameResponse {
  repeated FrameData frames = 1;
}

message FrameData {
  bytes image = 1;
  string timestamp = 2;
}
